Src/FluentAssertions/Types/MethodInfoSelector.cs

```diff
@@ -128,6 +128,17 @@ public MethodInfoSelector ThatAreNotDecoratedWithOrInherit<TAttribute>()
             return this;
         }
 
+        /// <summary>
+        /// Select return types of the methods
+        /// </summary>
+        /// <returns></returns>
+        public TypeSelector ReturnTypes()
```

Comment:
Can you give an example of why this would be useful?

Reply:
Sure. It's some kind of pseudocode

All of the methods that return collections should return immutable collections
`typeof(IDataReader).
        .Methods()
        .ReturnTypes()
        .ThatImplement<IEnumerable>()
        .Should()
        .BeImmutableCollection()`

All of the methods should return immutable types
`typeof(IDataReader).
        .Methods()
        .ReturnTypes()
        .Properties()
        .Should()
        .NotBeWritable()`